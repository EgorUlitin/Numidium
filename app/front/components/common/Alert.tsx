import * as React from 'react'

import { Alert } from 'antd'


export enum AlertType {
    SUCCESS = 'success',
    INFO    = 'info',
    WARNING = 'warning',
    ERROR   = 'error',
}

interface Props {
    type: AlertType
    message: string
}

export default (props: Props) => <Alert type={props.type} message={props.message} />
