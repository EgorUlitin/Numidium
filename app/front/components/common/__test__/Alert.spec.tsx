import * as React from 'react'
import * as renderer from 'react-test-renderer'

import Alert, { AlertType } from '../Alert'

test('alert renders correctly', () => {
    const tree = renderer
        .create(<Alert type={AlertType.INFO} message={'Test message!'} />)
        .toJSON()

    expect(tree).toMatchSnapshot()
})
