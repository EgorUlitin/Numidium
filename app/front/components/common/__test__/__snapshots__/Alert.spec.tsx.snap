// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Alert\` renders correctly 1`] = `
<div
  className="ant-alert ant-alert-info ant-alert-no-icon"
  data-show={true}
  style={undefined}
>
  <span
    className="ant-alert-message"
  >
    Test message!
  </span>
  <span
    className="ant-alert-description"
  />
</div>
`;
